nodewatcher/modules/monitor/http/survey/processors.py

```diff
@@ -0,0 +1,156 @@
+        # Should I go through the existing clients as well?
+        # existing_clients = {}
+        #for client in node.monitoring.network.clients():
+        #    existing_clients[client.client_id] = client
+
+        version = context.http.get_module_version("core.clients")
+        if version == 0:
+            # Unsupported version or data fetch failed (v0)
+            return context
+
+        channel = self.getNodeChannel(context, node)
+        snr = self.getNodeSNR(context, node)
+        neighbors = self.getNodeNeighbors(context, node)
+        print(neighbors)
+        if DATASTREAM_SUPPORTED:
+            # Store client count into datastream.
+            context.datastream.monitor_http_clients = SurveyInfoStreamsData(node, "2", channel, snr, neighbors)
+
+        return context
+
+    def getNodeChannel(self, context, node, frequency_band="2"):
+        """
+        Extracts the channel number from the context for the specified frequency band.
+        :param context: Current context
+        :param node: Node that is being processed
+        :param frequency_band: either "2" or "5", corresponding to 2.4GHz and 5GHz.
+        :return: channel currently assigned to the node at the specified frequency band.
+        """
+        try:
+            for interface in context.http.core.wireless.interfaces:
+                if str(context.http.core.wireless.interfaces[interface].frequency)[:1] == frequency_band:
+                    return context.http.core.wireless.interfaces[interface].channel
+        except NameError:
+            print("Error parsing JSON file for " + node)
+
+    def getNodeSNR(self, context, node, frequency_band="2"):
+        """
+        Extracts the SNR from the context for the specified frequency band.
+        :param context: Current context
+        :param node: Node that is being processed
+        :param frequency_band: either "2" or "5", corresponding to 2.4GHz and 5GHz.
+        :return: SNR of the node at the specified frequency band.
+        """
+        try:
+            for interface in context.http.core.wireless.interfaces:
+                if str(context.http.core.wireless.interfaces[interface].frequency)[:1] == frequency_band:
+                    signal = context.http.core.wireless.interfaces[interface].signal
+                    noise = context.http.core.wireless.interfaces[interface].noise
+                    if (not signal or not noise):
+                        print("No SNR available for " + node)
+                        return 0
+                    else:
+                        return signal - noise
+        except NameError:
+            print("Error parsing JSON file for " + node)
+        except KeyError:
+            print("No SNR available for " + node)
+            return 0
+
+    def getNodeNeighbors(self, context, node, frequency_band="2"):
+        """
+        returns a dictionary of all access points in the vicinity along with the signal strength of each access point.
```

Comment:
Documentation blocks fixed with 6834bb8

What length is considered long? My IDE wraps words at 120 characters.